### PR TITLE
Add two-bucket exercise

### PR DESCRIPTION
**Pull request by ryanplusplus**
See https://github.com/exercism/problem-specifications/tree/master/exercises/two-bucket

### CODE REVIEW

File: exercises/rational-numbers/README.md [hunk 1]
```diff
@@ -4,13 +4,13 @@ A rational number is defined as the quotient of two integers `a` and `b`, called
...
+The sum of two rational numbers `r₁ = a₁/b₁` and `r₂ = a₂/b₂` is `r₁ + r₂ = a₁/b₁ + a₂/b₂ = (a₁ * b₂ + a₂ * b₁) / (b₁ * b₂)`.
 
-The difference of two rational numbers `r1 = a1/b1` and `r2 = a2/b2` is `r1 - r2 = a1/b1 - a2/b2 = (a1 * b2 - a2 * b1) / (b1 * b2)`.
+The difference of two rational numbers `r₁ = a₁/b₁` and `r₂ = a₂/b₂` is `r₁ - r₂ = a₁/b₁ - a₂/b₂ = (a₁ * b₂ - a₂ * b₁) / (b₁ * b₂)`.
 
-The product (multiplication) of two rational numbers `r1 = a1/b1` and `r2 = a2/b2` is `r1 * r2 = (a1 * a2) / (b1 * b2)`.
+The product (multiplication) of two rational numbers `r₁ = a₁/b₁` and `r₂ = a₂/b₂` is `r₁ * r₂ = (a₁ * a₂) / (b₁ * b₂)`.
 
-Dividing a rational number `r1 = a1/b1` by another `r2 = a2/b2` is `r1 / r2 = (a1 * b2) / (a2 * b1)` if `a2 * b1` is not zero.
+Dividing a rational number `r₁ = a₁/b₁` by another `r₂ = a₂/b₂` is `r₁ / r₂ = (a₁ * b₂) / (a₂ * b₁)` if `a₂` is not zero.
```
Comment (wolf99) [inline exercises/rational-numbers/README.md:13]: Are these changes related to two-bucket?

File: exercises/two-bucket/.meta/tests.toml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+#"eb329c63-5540-4735-b30b-97f7f4df0f84" = true
+
+# Not possible to reach the goal
+#"449be72d-b10a-4f4b-a959-ca741e333b72" = true
+
+# With the same buckets but a different goal, then it is possible
+#"aac38b7a-77f4-4d62-9b91-8846d533b054" = true
+
+# Goal larger than both buckets is impossible
+#"74633132-0ccf-49de-8450-af4ab2e3b299" = true,
```
Comment (wolf99) [inline exercises/two-bucket/.meta/tests.toml:28]: I think the lines with the SHAs, should not have octothorpes prefixed as it comments out those line

**Review by wolf99 — CHANGES_REQUESTED** [review]
Few small comments, looks good overall though.

File: exercises/rational-numbers/README.md (hunk 1 above)
Comment (ryanplusplus) [reply exercises/rational-numbers/README.md:13]: Not directly, but when `configlet` is used to generate the READMEs it generates them for all exercises. This one was out of date.

File: exercises/two-bucket/.meta/tests.toml (hunk 2 above)
Comment (ryanplusplus) [reply exercises/two-bucket/.meta/tests.toml:28]: Oh got it, thanks!

File: exercises/two-bucket/src/example.h [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+#define TWO_BUCKET_H
+
+#include <stdbool.h>
+
+typedef enum {
+   bucket_id_1,
+   bucket_id_2
+} bucket_id_t;
+
+typedef unsigned bucket_volume_t;
```
[the remark this answers is not in the document]
Comment (ryanplusplus) [reply exercises/two-bucket/src/example.h:11]: :+1:

File: exercises/two-bucket/.meta/tests.toml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+"eb329c63-5540-4735-b30b-97f7f4df0f84" = true
+
+# Not possible to reach the goal
+"449be72d-b10a-4f4b-a959-ca741e333b72" = true
+
+# With the same buckets but a different goal, then it is possible
+"aac38b7a-77f4-4d62-9b91-8846d533b054" = true
+
+# Goal larger than both buckets is impossible
+"74633132-0ccf-49de-8450-af4ab2e3b299" = true,
```
Comment (patricksjackson) [inline exercises/two-bucket/.meta/tests.toml:28]: nit: Unnecessary comma?

File: exercises/two-bucket/src/example.h [hunk 5]
```diff
@@ -0,0 +1,24 @@
+#ifndef TWO_BUCKET_H
+#define TWO_BUCKET_H
+
+#include <stdbool.h>
+
+typedef enum {
+   bucket_id_1,
```
Comment (patricksjackson) [inline exercises/two-bucket/src/example.h:7]: nit: I believe we usually use ALL CAPS enums.

File: exercises/two-bucket/src/example.h [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+   bucket_id_2
+} bucket_id_t;
+
+typedef unsigned int bucket_volume_t;
+
+typedef struct {
+   bool possible;
+   int move_count;
+   bucket_id_t goal_bucket;
+   bucket_volume_t other_bucket_volume;
```
Comment (patricksjackson) [inline exercises/two-bucket/src/example.h:17]: The naming of `other_bucket_volume` and `goal_volume` is confusing to me. `volume` seems like a better descriptor for the size of the bucket and not the amount of water in the bucket.

Consider using something like `other_bucket_remaining_liters` or `goal_liters`.

File: exercises/two-bucket/test/test_two_bucket.c [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+}
+
+void tearDown(void)
+{
+}
+
+static void assert_results_match(bucket_result_t expected,
+                                 bucket_result_t actual)
+{
+   TEST_ASSERT_EQUAL(actual.possible, expected.possible);
```
Comment (patricksjackson) [inline exercises/two-bucket/test/test_two_bucket.c:15]: The order matters here, these tests should be `TEST_ASSERT_EQUAL(expected.whatever, actual.whatever)`

The Unity error message is "FAIL: Expected \<first argument\> Was \<second argument\>".

**Review by patricksjackson — APPROVED** [review]

File: exercises/two-bucket/.meta/tests.toml (hunk 4 above)
Comment (ryanplusplus) [reply exercises/two-bucket/.meta/tests.toml:28]: Whoops, good catch

File: exercises/two-bucket/src/example.h (hunk 5 above)
Comment (ryanplusplus) [reply exercises/two-bucket/src/example.h:7]: Yep, you're right. I'll get this fixed.

**Review by wolf99 — APPROVED** [review]
